presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSourceFactory.java

```diff
@@ -151,6 +151,7 @@ public int partitions()
     {
         lock.writeLock().lock();
         try {
+            checkState(!destroyed.isDone(), "PartitionedLookupSourceFactory has been destroyed");
```

Comment:
nit: shorter version would be more resilient to potential future renames

```suggestion
            checkState(!destroyed.isDone(), "already destroyed");
```